base/Base.jl

```diff
@@ -593,14 +593,19 @@ in_sysimage(pkgid::PkgId) = pkgid in _sysimage_modules
 if is_primary_base_module
 
 # Profiling helper
-# triggers printing the report and (optionally) saving a heap snapshot after a SIGINFO/SIGUSR1 profile request
+# triggers printing the report and (optionally) saving a heap snapshot/printing task backtraces
+# after a SIGINFO/SIGUSR1 profile request
 # Needs to be in Base because Profile is no longer loaded on boot
 function profile_printing_listener(cond::Base.AsyncCondition)
     profile = nothing
     try
         while _trywait(cond)
             profile = @something(profile, require_stdlib(PkgId(UUID("9abbd945-dff8-562f-b5e8-e1ebf5ef1b79"), "Profile")))::Module
             invokelatest(profile.peek_report[])
+            if Base.get_bool_env("JULIA_PROFILE_PEEK_TASK_BACKTRACES", false) === true
+                println(stderr, "Printing Julia task backtraces...")
+                ccall(:jl_print_task_backtraces, Cvoid, ())
```

Comment:
This prints to a different abstraction over stderr than the other println calls here, so may need to explicitly call flush around it to ensure the output doesn't get jumbled

Aside, it can also be better to use `print(stderr, "<text>\n")` instead of `println` where reliable async behavior is desired, since it ensures the whole text prints in the same syscall

Reply:
This is very helpful, thanks!